docs/architecture/decisions/0010-release-and-deployment-process-for-azure-functions.md

```diff
@@ -0,0 +1,70 @@
+# 10. Release and deployment process for Azure Functions
+
+Date: 2017-10-20
+
+## Status
+
+Accepted
+
+## Context
+
+The Digital Citizenship APIs are based on the Azure Functions framework (a _serverless_ application framework). 
+
+During the initial development of the APIs, we relied on the [continuous deployment feature](https://docs.microsoft.com/en-us/azure/azure-functions/functions-continuous-deployment), configured to deploy the API code from the `master` branch of the [digital-citizenship-functions](https://github.com/teamdigitale/digital-citizenship-functions) repository.
+
+This release process served us well during the initial development phase, but it has several drawbacks:
+
+1. there's no approval process that triggers the deployment of the `master` branch into production
+1. the build phase happens on the production hosts
+1. we cannot control the actual deployment process (the build and the re-load of the code happens automatically on the production hosts)
+1. Rolling back is cumbersome as it requires revering commits on the `master` branch.
+
+## Decision
+
+We decide to structure the release and deployment process in the following steps:
+
+### 1 - Automated tests
+
+On commit to the `master` branch, the CI system checks out the commit and runs the automated tests.
+
+### 2 - Build of release artifacts
+
+On tests passing successfully, the CI workflow triggers the build process that generates artifacts needed for release (e.g. packs dependencies, compiles Typescript into Javascript, etc...).
+
+The generated artifact get stored in a `build` directory (e.g. a single javascript file generated by [Azure Functions Pack](https://github.com/Azure/azure-functions-pack)).
+
+### 3 - Version bump and tags
+
+On successful build, the patch version gets incremented (e.g. in `package.json`) and it gets committed along with the generated artifacts. 
+
+After commit, two tags get created:
+
+* a `SNAPSHOT-x.x.x` tag using the bumped version
+* a `SNAPSHOT-LATEST` tag (this tag gets updated to the current commit)  
+
+Everything gets pushed to the upstream repository.
+
+### 4 - Promotion
+
+On the CI workflow, a promotion step awaits for approval.
```

Comment:
it's not clear how

Reply:
yes sorry, I should add links, I'm talking about [circleci workflows](https://circleci.com/docs/2.0/workflows/)